### PR TITLE
Add queryFilter to triggers

### LINKED ISSUE #13746: Add `queryFilter` to the daml-trigger library
This should match the behavior of the `queryFilter` function from Daml script.

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin

- [Daml Triggers] Add `queryFilter` matching Daml Script’s
  `queryFilter` which queries contracts of a given template and filters
  them down to those where the predicate holds.

changelog_end

fixes #13746

### CODE REVIEW

File: triggers/daml/Daml/Trigger.daml [hunk 1]
```diff
@@ -84,10 +85,16 @@ getContractById id (ACS tpls pending) = do
   a <- fromAnyTemplate aa
   if any (elem aid) pending then None else Some a
 
+
```
Comment (remyhaemmerle-da) [inline triggers/daml/Daml/Trigger.daml:88]: Why an extra line here ?

**Review by remyhaemmerle-da — APPROVED** [review]
looks reasonable.

File: triggers/daml/Daml/Trigger.daml (hunk 1 above)
Comment (cocreature) [inline triggers/daml/Daml/Trigger.daml:88]:
```suggestion
```

Comment (cocreature) [reply triggers/daml/Daml/Trigger.daml]: because I’m stupid :)

**Pull request merged**